### ISSUE TITLE
Differentiate between messages coming from render and main process?

### ISSUE THREAD

**Issue by declanvk** [megahertz/electron-log#75]
Is it possible to differentiate the log format for messages coming from the main process versus the render process?

For instance:
```
[10:39:18.216] [main] [info] In development mode
[10:39:18.219] [main] [info] Logging to console at 'info' level
[10:39:25.577] [render] [info] Starting WebSocket: ws://localhost:8888/api/kernels/740e543d-70f4-412d-8efc-f982af7d3c88
[10:39:25.578] [render] [info] Starting WebSocket: ws://localhost:8888/api/kernels/740e543d-70f4-412d-8efc-f982af7d3c88
```

I attempted to do this via the `log.transports.console.format` and the `log.transports.rendererConsole.format` options, but it only formatted it using the `log.transports.console.format`.

**Comment by megahertz** [comment]
It's not possible now, but it will be possible in v3.

**Comment by aguynamedben** [comment]
One thing to note for this idea: Sometimes apps will have many processes, and it would be nice to differentiate between multiple renderer processes. For example, my app has 3 processes:
* main
* renderer
* backend

The `backend` process is really another renderer with a BrowserWindow that never .show()'s. I make RPC calls to it to do data-intensive work I don't want happening in the main or renderer processes (i.e. don't lock up UI rendering!)

So I would like my logs to look like...
```
[10:39:18.216] [main] [info] In development mode
[10:39:18.219] [main] [info] Logging to console at 'info' level
[10:39:25.577] [render] [info] Starting WebSocket: ws://localhost:8888/api/kernels/740e543d-70f4-412d-8efc-f982af7d3c88
[10:39:25.578] [render] [info] Starting WebSocket: ws://localhost:8888/api/kernels/740e543d-70f4-412d-8efc-f982af7d3c88
[10:39:18.216] [backend] [info] In development mode
[10:39:18.219] [backend] [info] Logging to console at 'info' level
```

I presume the easiest way for electron-log to differentiate might be to check an ENV variable I set? i.e. process.ENV.PROCESS_NAME.  Maybe there's a more accepted way.

**Comment by bhj** [comment]
Sort of related, I have a forked process that should write to its own file. It would be nice if you could set `log.transports.file.filename` to keep the auto path detection and just use a different filename.

**Comment by megahertz** [comment]
@bhj I'll implement a filename property in v3.

**Comment by megahertz** [comment]
Resolved in v3.0.0-beta2

**Comment by aguynamedben** [comment]
Sweet, I'll check it out, thanks so much! 👏
